To: vim_dev@googlegroups.com
Subject: Patch 7.3.263
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.263
Problem:    Perl and Tcl have a few code style problems.
Solution:   Clean it up.  (Elias Diem)
Files:	    src/if_perl.xs, src/if_tcl.c


*** ../vim-7.3.262/src/if_perl.xs	2010-10-23 14:02:48.000000000 +0200
--- src/if_perl.xs	2011-07-27 14:11:03.000000000 +0200
***************
*** 755,761 ****
  #ifdef HAVE_SANDBOX
      if (sandbox)
      {
! 	safe = perl_get_sv( "VIM::safe", FALSE );
  # ifndef MAKE_TEST  /* avoid a warning for unreachable code */
  	if (safe == NULL || !SvTRUE(safe))
  	    EMSG(_("E299: Perl evaluation forbidden in sandbox without the Safe module"));
--- 755,761 ----
  #ifdef HAVE_SANDBOX
      if (sandbox)
      {
! 	safe = perl_get_sv("VIM::safe", FALSE);
  # ifndef MAKE_TEST  /* avoid a warning for unreachable code */
  	if (safe == NULL || !SvTRUE(safe))
  	    EMSG(_("E299: Perl evaluation forbidden in sandbox without the Safe module"));
***************
*** 1108,1114 ****
      VIWIN win
  
      PPCODE:
!     if(items == 1)
      {
        EXTEND(sp, 2);
        if (!win_valid(win))
--- 1108,1114 ----
      VIWIN win
  
      PPCODE:
!     if (items == 1)
      {
        EXTEND(sp, 2);
        if (!win_valid(win))
***************
*** 1116,1122 ****
        PUSHs(sv_2mortal(newSViv(win->w_cursor.lnum)));
        PUSHs(sv_2mortal(newSViv(win->w_cursor.col)));
      }
!     else if(items == 3)
      {
        int lnum, col;
  
--- 1116,1122 ----
        PUSHs(sv_2mortal(newSViv(win->w_cursor.lnum)));
        PUSHs(sv_2mortal(newSViv(win->w_cursor.col)));
      }
!     else if (items == 3)
      {
        int lnum, col;
  
***************
*** 1249,1257 ****
  	{
  	    lnum = SvIV(ST(1));
  	    count = 1 + SvIV(ST(2)) - lnum;
! 	    if(count == 0)
  		count = 1;
! 	    if(count < 0)
  	    {
  		lnum -= count;
  		count = -count;
--- 1249,1257 ----
  	{
  	    lnum = SvIV(ST(1));
  	    count = 1 + SvIV(ST(2)) - lnum;
! 	    if (count == 0)
  		count = 1;
! 	    if (count < 0)
  	    {
  		lnum -= count;
  		count = -count;
*** ../vim-7.3.262/src/if_tcl.c	2010-12-17 20:23:56.000000000 +0100
--- src/if_tcl.c	2011-07-27 14:11:03.000000000 +0200
***************
*** 1884,1897 ****
  {
      int newerr = OK;
  
!     if (error == TCL_EXIT )
      {
  	int retval;
  	char buf[50];
  	Tcl_Obj *robj;
  
  	robj = Tcl_GetObjResult(tclinfo.interp);
! 	if( Tcl_GetIntFromObj(tclinfo.interp, robj, &retval) != TCL_OK )
  	{
  	    EMSG(_("E281: TCL ERROR: exit code is not int!? Please report this to vim-dev@vim.org"));
  	    newerr = FAIL;
--- 1884,1897 ----
  {
      int newerr = OK;
  
!     if (error == TCL_EXIT)
      {
  	int retval;
  	char buf[50];
  	Tcl_Obj *robj;
  
  	robj = Tcl_GetObjResult(tclinfo.interp);
! 	if (Tcl_GetIntFromObj(tclinfo.interp, robj, &retval) != TCL_OK)
  	{
  	    EMSG(_("E281: TCL ERROR: exit code is not int!? Please report this to vim-dev@vim.org"));
  	    newerr = FAIL;
***************
*** 1900,1906 ****
  	{
  	    sprintf(buf, _("E572: exit code %d"), retval);
  	    tclerrmsg(buf);
! 	    if (retval == 0 )
  	    {
  		did_emsg = 0;
  		newerr = OK;
--- 1900,1906 ----
  	{
  	    sprintf(buf, _("E572: exit code %d"), retval);
  	    tclerrmsg(buf);
! 	    if (retval == 0)
  	    {
  		did_emsg = 0;
  		newerr = OK;
*** ../vim-7.3.262/src/version.c	2011-07-27 14:09:05.000000000 +0200
--- src/version.c	2011-07-27 14:12:05.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     263,
  /**/

-- 
DEAD PERSON:  I don't want to go in the cart!
CUSTOMER:     Oh, don't be such a baby.
MORTICIAN:    I can't take him...
DEAD PERSON:  I feel fine!
CUSTOMER:     Oh, do us a favor...
MORTICIAN:    I can't.
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
